src/main/java/mate/academy/spring/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(() ->
+                new UsernameNotFoundException("Can`t find user with username: " + username));
+        UserBuilder userBuilder = withUsername(username);
+        userBuilder.password(user.getPassword());
+        userBuilder.authorities(user.getRoles().stream()
```

Comment:
you can use roles() instead authorities() for more consistency (in this case you also should remove prefix ROLE from role names enum):
```suggestion
        userBuilder.roles(user.getRoles().stream()
```